osquery/tables/events/darwin/yara.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ *  Copyright (c) 2015, Wesley Shields
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <map>
+#include <string>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+
+#include <yara.h>
+
+#include "osquery/events/darwin/fsevents.h"
+
+namespace osquery {
+namespace tables {
+
+/**
+ * @brief Track YARA matches to files.
+ */
+class YARAEventSubscriber : public EventSubscriber<FSEventsEventPublisher> {
+  DECLARE_SUBSCRIBER("yara");
+
+ public:
+  void init();
+
+ private:
+  // XXX: Is there a better way to say "I'm not ready to receive events"?
```

Comment:
Ok! We should change `init()` to return a status, if it returns a failure all subscriptions the subscriber may have created would be removed. There's nothing like that now, but we could extend the events/pubsub API no problem.

We could also consider a general pause/resume API if you want to start some work thread in `init()` or abort in a callback while you cleanup.